src/core/network/webhook.ts

```diff
@@ -8,43 +8,36 @@ export default function (
   updateHandler: (update: Update, res: http.ServerResponse) => Promise<void>,
   errorHandler: (err: SyntaxError) => void
 ) {
-  return (
+  return async (
     req: http.IncomingMessage,
     res: http.ServerResponse,
-    next?: () => void
-  ): void => {
-    debug('Incoming request', req.method, req.url)
-    if (req.method !== 'POST' || req.url !== hookPath) {
-      if (typeof next === 'function') {
-        return next()
-      }
+    next = (): void => {
       res.statusCode = 403
       return res.end()
     }
+  ): Promise<void> => {
+    debug('Incoming request', req.method, req.url)
+    if (req.method !== 'POST' || req.url !== hookPath) {
+      return next()
+    }
     let body = ''
-    req.on('data', (chunk: string) => {
-      body += chunk.toString()
-    })
-    req.on('end', () => {
-      let update: Update
-      try {
-        update = JSON.parse(body)
-      } catch (error) {
-        res.writeHead(415)
-        res.end()
-        return errorHandler(error)
-      }
-      updateHandler(update, res)
-        .then(() => {
-          if (!res.finished) {
-            res.end()
-          }
-        })
-        .catch((err: unknown) => {
-          debug('Webhook error', err)
-          res.writeHead(500)
-          res.end()
-        })
-    })
+    for await (const chunk of req) {
+      body += String(chunk)
+    }
+    let update: Update
+    try {
+      update = JSON.parse(body)
+    } catch (error) {
+      res.writeHead(415)
+      res.end()
+      return errorHandler(error)
```

Comment:
One more thing, `Telegraf` passes its `debug` instance here, which is weird. I'll use This module's own one.